src/odemis/acq/fastem.py

```diff
@@ -276,6 +277,31 @@ def run(self):
         :raise:
             Exception: If it failed before any single field images were acquired or if acquisition was cancelled.
         """
+        self._beamshift.shift.value = (0, 0)
+
+        # Update the position of the first tile.
+        self._pos_first_tile = self.get_pos_first_tile()
+
+        # Move the stage to the first tile, to ensure the autofocus and image
+        # translation pre-align are done in the correct location.
+        self.move_stage_to_next_tile()
+
+        # Autofocus multiprobe ensures the beams are focused at the start of the ROA acquisition.
+        autofocus_multiprobe.run_autofocus(self._scanner,
```

Comment:
I wonder if we need a try except around this and the run_image_translation_pre_align.

Reply:
I have the feeling we should do that. If the calibration fails, it is nice if the exception is also handled by the caller as for the ROA acquisition I guess and also that we can put the system back into a "good state" with a finally statement. If I am not mistaken the exception for the ROA is handled by the callback here and that would put the GUI also back into a "good state".
```
    def on_acquisition_done(self, future):
        """
        Callback called when the acquisition is finished (either successfully or
        cancelled)
        """

        try:
            future.result()
            self._reset_acquisition_gui()
        except CancelledError:
            self._reset_acquisition_gui()
            return
        except Exception as exp:
            # leave the gauge, to give a hint on what went wrong.
            logging.exception("Acquisition failed")
            self._reset_acquisition_gui("Acquisition failed (see log panel).", level=logging.WARNING)
            return
```